sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,516 @@
+      &aon_timer, kDifAonTimerIrqWdogTimerBark,
+      &is_pending));  // (this wants "bool" not "volatile bool")
+  irq_is_pending = is_pending;
+
+  // Stop the watchdog.
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon_timer));
+
+  // In order to handle the NMI we need to acknowledge the interrupt status bit
+  // at the peripheral side. Note that this test does not use the PLIC, so there
+  // is nothing to acknowledge on the PLIC side. We are hence not using the
+  // isr_testutils_aon_timer_isr function here.
+  CHECK_DIF_OK(
+      dif_aon_timer_irq_acknowledge(&aon_timer, kDifAonTimerIrqWdogTimerBark));
+}
+
+//==============================================================================
+bool test_main(void) {
+  // ------------------------------------
+  // Define access to DUT IPs:
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR),
+      &aon_timer));  // AON Timer
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));  // IBEX core
+  CHECK_DIF_OK(
+      dif_pwrmgr_init(mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR),
+                      &pwrmgr));  // PWRMGR
+  CHECK_DIF_OK(
+      dif_rv_timer_init(mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR),
+                        &rv_timer));  // RV Timer
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));  //
+  CHECK_DIF_OK(dif_pwm_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWM_AON_BASE_ADDR), &pwm));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR),
+      &otp_ctrl));  // OTP Controller
+  CHECK_DIF_OK(
+      dif_gpio_init(mmio_region_from_addr(TOP_EARLGREY_GPIO_BASE_ADDR), &gpio));
+
+  LOG_INFO("SW_INFO: Running CHIP Power Idle Load test");
+
+  // ------------------------------------
+  // Testplan: "The test should set a GPIO (mapped to the IOA2 pin) to high
+  // while the power state of interest is active"
+
+  static const uint32_t kGpioMask = 0x00000004;  // use pin 2
+  //                                            // "If the Nth bit is enabled,
+  //                                            then the Nth pin is selected by
+  //                                            the mask"
+
+  // PINMUX -   // Set a connection between a MIO pad output and peripheral
+  // output
+  CHECK_DIF_OK(dif_pinmux_output_select(
+      &pinmux, (kTopEarlgreyPinmuxMioOutIoa0 + 2),
+      (kTopEarlgreyPinmuxOutselGpioGpio0 + 2)));  // pinmux config
+
+  // Set output modes of all GPIO pins
+  CHECK_DIF_OK(dif_gpio_output_set_enabled_all(&gpio, kGpioMask));
+
+  // Write to set IOA2 low at the start of the test:
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
```

Comment:
do you want to add a different vseq to check that the gpio actually behaves as you expect? You can do some logging and expectation checking.  This is not super critical though.

Reply:
My point was for the test to be self contained, i.e. without a host.
Do you want me to add a host VSEQ with this check?

Reply:
Agree with @tjaychen - addition of host component for DV strengthens the checks. The C test can still be run on other non-DV platforms.

Reply:
Done